src/components/CompareResults/ResultsTable.tsx

```diff
@@ -103,9 +103,34 @@ export default function ResultsTable() {
   const initialSearchTerm = rawSearchParams.get('search') ?? '';
   const [searchTerm, setSearchTerm] = useState(initialSearchTerm);
   const [frameworkIdVal, setFrameworkIdVal] = useState(frameworkId);
-  const [tableFilters, setTableFilters] = useState(
-    new Map() as Map<string, Set<string>>, // ColumnID -> Set<Values to remove>
-  );
+
+  const [tableFilters, setTableFilters] = useState(() => {
+    const filters = new Map();
+    for (const [key, value] of searchParams.entries()) {
+      if (key.startsWith('filter_')) {
+        const columnId = key.replace('filter_', '');
+        filters.set(columnId, new Set(value.split(',')));
+      }
+    }
+    return filters;
+  });
+
+  useEffect(() => {
+    const newSearchParams = new URLSearchParams(searchParams);
+    tableFilters.forEach((filterSet: Set<string>, columnId: string) => {
+      if (filterSet.size > 0) {
+        newSearchParams.set(
+          `filter_${columnId}`,
+          Array.from(filterSet).join(','),
+        );
+      } else {
+        newSearchParams.delete(`filter_${columnId}`);
+      }
+    });
+    if (newSearchParams.toString() !== searchParams.toString()) {
+      updateRawSearchParams(newSearchParams);
+    }
+  }, [tableFilters, searchParams, setSearchParams]);
 
```

Comment:
Good job on this first commit! I suggest memoizing `URLSearchParams` (see [useMemo ](https://react.dev/reference/react/useMemo)for react) to avoid recreating it on every render. You can then add it the `useEffect`'s dependency array to make sure it's only recalculated when `searchParams` changes.